src/Chapter2.hs

```diff
@@ -857,7 +885,20 @@ and reverses it.
   function, but in this task, you need to implement it manually. No
   cheating!
 -}
-rewind = error "rewind: Not Implemented!"
+
+-- I guess I miss something...
+rewind' :: [a] -> [a]
+rewind' []       = []
+rewind' (x : xs) = rewind xs ++ [x]
+
+-- this version prepends to target list so it may be more efficient
```

Comment:
You are absolutely right!

In lists, it is quite slow to add anything at the end of the list. That is why it is always better to rewrite it with the `:` cons. Remember the explanation with trains? 🚂 🚋 🚋 

That is why a more efficient solution is with the accumulator and the recursive function that will do the addition at the start of the list which is instant!

You can read a bit more about the `go` pattern in here if interested: https://kowainik.github.io/posts/haskell-mini-patterns#recursive-go